src/main/java/org/wildfly/security/auth/realm/ldap/LdapSecurityRealm.java

```diff
@@ -147,17 +150,29 @@ public void registerIdentityChangeListener(Consumer<Principal> listener) {
         DirContext dirContext = null;
         try {
             dirContext = obtainContext();
-            EventContext eventContext = (EventContext) dirContext.lookup("");
-            eventContext.addNamingListener("", EventContext.SUBTREE_SCOPE, new ServerNotificationListener(listener));
+            registerIdentityChangeListener(dirContext,listener);
```

Comment:
Won't there be a potential race like this:

Thread 1 | Thread 2
------|------
registerIdentityChangeListener fails|
-|connection succeeds
-|listeners pending registration added (empty)
listener added to pending registration list|

And then the listener is never called?